File: src/python/Regexp2.py
import sys
import re

def main():
    f = open(sys.argv[1])
    lines = f.readlines()
    f.close()

    if len(sys.argv) > 2 and sys.argv[2] == '--dry-run':
        for line in lines:
            pass
        return
    if len(sys.argv) > 3 and sys.argv[3] == '--dry-run':
        for line in lines:
            pass
        return

    #Pre-compile regexp before comparing to it
    #TODO: this should be excluded from "empty"-run
    if len(sys.argv) > 2 and sys.argv[2] == '--with-or-operator':
        pattern = re.compile(r"\d+|\w+\d+\.")
    else:
        pattern = re.compile(r"\d+")

    amountFound = 0
    for line in lines:
        match = pattern.search(line)
        if match is not None:
            amountFound = amountFound + 1

if __name__ == '__main__':
    main()

File: src/java/Regexp2.java
import java.io.*;
import java.util.List;
import java.util.ArrayList;
import java.util.regex.*;

class Regexp2 {
    public static void main(String[] args) {
        List<String> lines = new ArrayList<String>();
        try {
          BufferedReader br= new BufferedReader(new FileReader(args[0]));
          String str;
          while ((str = br.readLine()) != null) {
              lines.add(str + '\n');
          }
          br.close();
        } catch (Exception e) { System.out.println(e); }

        if (args.length > 1 && args[1].equals("--dry-run")) {
            return;
        }
        if (args.length > 2 && args[2].equals("--dry-run")) {
            return;
        }

        Pattern pattern = Pattern.compile("\\d+");
        if (args.length > 1 && args[1].equals("--with-or-operator")) {
            pattern = Pattern.compile("\\d+|\\w+\\d+\\.");
        }

        Matcher match;
        int amountFound = 0;
        for (int i=0; i < lines.size(); i++) {
            match = pattern.matcher(lines.get(i));
            if (match.find()) {
                amountFound++;
            }
        }

        //TODO: add cpu time measurement
    }
}
